rfd/0133-connect-my-computer.md

```diff
@@ -0,0 +1,513 @@
+
+The subsequent launches of the agent follow a similar logic to that of the agent launch during the
+setup. The only difference is that on subsequent launches Connect doesn’t have to generate the
+config file and the join token. Instead, it can verify that the agent binary exists and then attempt
+to launch it.
+
+After launching the agent, Connect flips a per-workspace flag `autoStart` to `true`. This means that
+if the user closes the app while the agent is running, on the next app launch Connect is going to
+automatically start the agent on a successful login to the cluster. The flag is flipped to `false`
+if the user explicitly stops the agent through the UI – just closing the app will not change the
+flag state.
+
+### Agent maintenance
+
+#### Keeping the agent up to date
+
+Upgrading the agent is done by upgrading Connect. On the first agent launch after a Connect upgrade,
+Connect sees that the agent matching its version is missing and downloads a matching version of the
+agent. This follows from the behavior described in the “Downloading the agent” section.
+
+#### Log rotation
+
+The MVP is not going to perform any extra log rotation beyond what the Teleport agent offerts out of
+the box which is creating a new log file every 24 hours. The logs are thus stored in the same
+directory as the data directory of the agent and are removed together with the agent.
+
+In the future we might consider adjusting `teleport node configure` so that `teleport.log.output`
+can be set from the CLI.
+
+### Removing the agent
+
+The agent can be removed completely by clicking on the “Remove agent” button on the status tab or
+partially by logging out.
+
+#### By clicking “Remove agent” on the status tab
+
+This removes the config and the data dir together with any CMC-related app state and an attempt is
+made to remove the CMC role from the cluster. The agent binary is kept in the sessionData folder to
+enable a scenario where the user wants to remove the agent from one cluster but wants to keep using
+it in another cluster.
+
+Removing the agent performs the following actions:
+
+1. Stop the agent process for the given cluster.
+2. Capture the UUID of the node for use in subsequent steps.
+3. Remove the agent data at `<userData>/agents/<profile>`.
+4. Call [`DeleteNode`](https://github.com/gravitational/teleport/blob/7e174f9aaf17d0f3ea08740aac42da49eb7ec8da/api/client/client.go#L2044)
+   with the node UUID to remove the node from the cluster.
+
+    The request may fail if the user does not have permission to remove nodes. This will not be
+    considered a fatal error. Instead, at the end of the agent removal process Connect will explain
+    to the user that the node can still be visible for a few minutes until it is purged from the
+    cluster cache. This should help prevent user confusion regarding the continued visibility of the
+    CMC node, as they may not be fully aware of how the cluster cache works.
+
+5. Remove the CMC role.
```

Comment:
@gzdunek @zmb3 I need feedback on the above regarding removing the role together with the agent.

While working on setting up the role, I noticed that removing a role is a highly destructive operation. When I want to clean up and restart my role-creating code, I'm locked out of the Web UI as soon as I remove the role from the cluster. That's because the cert that my browser session uses has in it the role that I just removed. So the cluster responds with an error when it tries to check my permissions for doing any action in the cluster.

* Should we attempt to remove the Connect My Computer role when removing the agent?
* Does the MVP need to support a case where one cluster user wants to use Connect My Computer on two different devices with a different username?
   * This is basically asking if we're allowed to cut corners here. 🤷

Reply:
Given how destructive removing a role is, I feel like we should just not do this.

Imagine you've been using CMC for a couple of days and now you want to remove it. You're logged in Connect but also in your global tsh or in the Web UI. You remove an agent and suddenly your existing session outside of Connect stops working and you have to manually relogin.

That's much worse UX than just having one extra role left on your cluster which you're not going to use.

Reply:
Ah right, I forgot that we create `connect-my-computer-<cluster-username>`, not `connect-my-computer-<login>` role. I'm wondering if using the `connect-my-computer-<cluster-username>-<login>` scheme wouldn't solve this problem quickly? (I can't recall if we considered it)

Reply:
Yeah, it's only safe to remove the role when there are no existing credentials out there with said role.

I think we can just leave it behind for now. Especially if we can ensure that we re-use the role if the user decides to turn on CMC again in the future.

Reply:
Ok, I'm going to update the RFD so that:

* We keep the role.
* When setting up CMC, we make sure that the role contains the current system username. If not, we add it to logins in the role.

There's an existing issue for improving the UX after role removal (#8835).